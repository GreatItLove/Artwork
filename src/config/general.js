export const IMAGE_BASE_PATH = 'http://www.artworkmanagerdemo.com/images/';

export const SYS_MESSAGE_TYPE_ERROR = 'error';
export const SYS_MESSAGE_TYPE_SUCCESS = 'success';
export const SYS_MESSAGE_TYPE_INFO = 'info';
