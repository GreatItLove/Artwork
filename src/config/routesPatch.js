export const ADMIN_PANEL_BASE_PATH = '/';
export const ADMIN_PANEL_BASE_PATH_PREFIX = '';
export const LOGIN_PATH = '/login';
export const FORGOT_PASSWORD_PATH = '/forgot-password';
export const RESET_PASSWORD_PATH = '/reset-password';

//Resource path
export const ARTISTS_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/artists`;
export const ARTISTS_EDIT_PATH = `${ARTISTS_PATH}/edit`;
export const ARTWORK_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/artwork`;
export const CONSIGNMENTS_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/consignments`;
export const CONTACTS_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/contacts`;
export const INVOICES_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/invoices`;
export const REPORTS_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/reports`;
export const MARKETING_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/marketing`;
export const WEBSITES_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/website`;
export const MEDIA_LIBRARY = `${ADMIN_PANEL_BASE_PATH_PREFIX}/media-library`;

//Other admin path
export const EXIBITIONS_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/exhibitions`;
export const RESOURCES_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/resources`;
export const QUICK_SALE_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/quick-sale`;
export const VISITOR_TODAY_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/visitor-today`;
export const SURVEY_RESULTS_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/survey-results`;

//Account
export const ACCOUNT_EDIT_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/account/edit`;

//Users resource
export const USERS_LIST_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/users`;
export const USERS_RESET_PASSWORD_PATH = `${ADMIN_PANEL_BASE_PATH_PREFIX}/users/reset-password`;
