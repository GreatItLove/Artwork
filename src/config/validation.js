export const MAX_EMAIL_LENGTH = 50;
export const MAX_FULL_NAME_LENGTH = 50;
