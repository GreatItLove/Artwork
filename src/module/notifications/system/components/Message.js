import React, { Component } from 'react';
import enhanceWithClickOutside from 'react-click-outside';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import { SYS_MESSAGE_TYPE_ERROR, SYS_MESSAGE_TYPE_SUCCESS, SYS_MESSAGE_TYPE_INFO } from '../../../../config/general';

const propTypes = {
  removeMessage: PropTypes.func.isRequired,
  messageType: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired
};

/**
 * @class Message
 * @extends React.Component
 * @description Render component
 */
class Message extends Component {
  componentDidMount() {
    this.timeOutID = setTimeout(() => {
      this.removeMesage();
    }, 6000);
  }
  componentWillUnmount() {
    clearTimeout(this.timeOutID);
  }

  handleClickOutside() {
    this.removeMesage();
  }
  removeMesage = () => {
    this.props.removeMessage();
  };
  /**
   * @description render
   * @return {Object} JSX HTML Content
   */
  render() {
    const { message, messageType } = this.props;
    const alertClassNames = classnames('alert', {
      'alert-danger': messageType === SYS_MESSAGE_TYPE_ERROR,
      'alert-success': messageType === SYS_MESSAGE_TYPE_SUCCESS,
      'alert-info': messageType === SYS_MESSAGE_TYPE_INFO
    });
    return (
      <div className={alertClassNames} role="alert">
        <button onClick={this.removeMesage} type="button" className="close" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
        {message}
      </div>
    );
  }
}
Message.propTypes = propTypes;

export { Message as MessageTest };

export default enhanceWithClickOutside(Message);
