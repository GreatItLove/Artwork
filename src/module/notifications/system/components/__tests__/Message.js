import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { MessageTest as Message } from '../Message';
import { SYS_MESSAGE_TYPE_ERROR } from '../../../../../config/general';

describe('Message', () => {
  it('Message error snapshot', () => {
    const removeMessage = jest.fn();
    const messageType = SYS_MESSAGE_TYPE_ERROR;
    const message = 'some error message';
    const wrapper = shallow(<Message {...{ removeMessage, messageType, message }} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
