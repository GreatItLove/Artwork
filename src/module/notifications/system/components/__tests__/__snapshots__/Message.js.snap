// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Message error snapshot 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  <button
    aria-label="Close"
    className="close"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
  </button>
  some error message
</div>
`;
